go-controller/pkg/util/multi_network.go

```diff
@@ -822,6 +822,11 @@ func (nInfo *secondaryNetInfo) canReconcile(other NetInfo) bool {
 	if nInfo == nil && other == nil {
 		return true
 	}
+	// if network ID has changed, it means the network was re-created, and all controllers
+	// should execute delete+create instead of update
+	if nInfo.GetNetworkID() != types.InvalidID && other.GetNetworkID() != types.InvalidID && nInfo.GetNetworkID() != other.GetNetworkID() {
```

Comment:
I don't think we should ignore the case of going from a valid id to invalid one.
```
if nInfo.GetNetworkID() != types.InvalidID && nInfo.GetNetworkID() != other.GetNetworkID() {
```
should be enough imo.

Reply:
> That makes sense, but technically canReconcile doesn't say which one is old and which is new.

Although we use it this way that is a good point, `AreNetworksCompatible` doesn't differentiate new and current technically.
 
IIUC it is hard to get a `InvalidID` with UDN because we are now setting the ID as an annotation on the NAD when it is being created. 
But in general if a network switches IDs I would expect for it to be recreated, even if the it switches to/from is set to `InvalidID`.